src/main/scala/com/linkedin/feathr/offline/job/FeatureGenJob.scala

```diff
@@ -174,9 +175,57 @@ object FeatureGenJob {
     withParamsOverrideConfig.root().render()
   }
 
+  /**
+   * Load the DataFrames for sources that needs preprocessing by Pyspark.
+   * @param args Same arguments for the main job.
+   * @param featureNamesInAnchorSet A set of feature names of an anchor sorted and joined by comma. For example,
+   *                                anchor1 -> f1, f2, anchor2 -> f3, f4. Then the set is ("f1,f2", "f3,f4")
+   * @return A Java map whose key is Feature names of an anchor sorted and joined by comma and value is the DataFrame
+   *         for this anchor source. For example, anchor1 -> f1, f2, anchor2 -> f3, f4. Then the result is
+   *         Map("f1,f2" -> df1, "f3,f4" -> df2).
+   */
+  def loadSourceDataframe(args: Array[String], featureNamesInAnchorSet: java.util.Set[String]): java.util.Map[String, DataFrame] = {
```

Comment:
Can we unify the two loadSourceDataframe ?

Reply:
I feel it' a bit hard with current setup so i would avoid major refactoring at the moment. Let me know if you have some good ideas.